From b2e5813935f6897bd3a2c7ee5be4a3db17f58745 Mon Sep 17 00:00:00 2001
From: Jason Gerecke <killertofu@gmail.com>
Date: Mon, 1 Aug 2011 17:51:00 -0700
Subject: [PATCH] xsetwacom: Have 'get_param' respect PROP_FLAG_INVERTED

Running e.g. `xsetwacom get stylus TabletPCButton` would display
the wrong value to the user because the property is inverted.
This checks to see if PROP_FLAG_INVERTED is set, and to swap the
result if so.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 tools/xsetwacom.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/tools/xsetwacom.c b/tools/xsetwacom.c
index 365bad4..86bf714 100644
--- a/tools/xsetwacom.c
+++ b/tools/xsetwacom.c
@@ -2230,7 +2230,10 @@ static void get_param(Display *dpy, XDevice *dev, param_t *param, int argc, char
 				int val = data[param->prop_offset + i];
 
 				if (param->prop_flags & PROP_FLAG_BOOLEAN)
-					sprintf(&str[strlen(str)], "%s", val ?  "on" : "off");
+					if (param->prop_flags & PROP_FLAG_INVERTED)
+						sprintf(&str[strlen(str)], "%s", val ?  "off" : "on");
+					else
+						sprintf(&str[strlen(str)], "%s", val ?  "on" : "off");
 				else
 					sprintf(&str[strlen(str)], "%d", val);
 
-- 
1.7.4.1

